src/Core/Managed/Shared/Channel/InMemoryChannel.cs

```diff
@@ -133,7 +143,14 @@ public void Flush()
         /// <param name="timeout">Timeout interval to abort sending.</param>
         public void Flush(TimeSpan timeout)
         {
-            this.transmitter.Flush(timeout);
+            if (!isDisposed)
+            {
+                this.transmitter.Flush(timeout);
+            }
+            else
+            {
+                throw new ObjectDisposedException("InMemoryChannel", "The telemetry channel has been disposed.");
```

Comment:
I think this logic should live in the `InMemoryTransmitter`; the `InMemoryChannel` only assumes that `InMemoryTransitter.Flush` is invalid after `InMemoryTransmitter.Dispose` but it's the responsibility of the `InMemoryTransmitter` to enforce that.